### ISSUE TITLE
Errors with ggsurvplot and ggcoxadjustedcurves functions

### ISSUE THREAD

**Issue by DanaiChas** [kassambara/survminer#280]
When I’m running:  ggsurvplot(survfit(cox.train32, newdata=data.train.freq.demo)

I’m getting the following: 

Error in .get_data(fit, data = data, complain = FALSE) : 
  The `data` argument should be provided either to ggsurvfit or survfit.

When I’m running: ggcoxadjustedcurves(cox, data = data1, individual.curves = TRUE)

I’m getting the following:  Error: could not find function “ggcoxadjustedcurves"

My goal is to get survival curves for each individual in the dataset

Because ggsurvplot (survfit(Surv(time,Status) ~ . , newdata=data1), adata=data1)

gives an error: Error in terms.formula(formula, data = data) :   '.' in formula and no 'data’ argument

,  would the following work?

cox <- coxph(Surv(time,Status) ~ .  , data=data)
ggsurvplot(survfit(cox, newdata=data1), data=data1)

For the ggcoxadjustedcurves, I was using individual.curves = TRUE to get individual survival curves.

How can I get the same result using the new function ggadjustedcurves?

**Comment by kassambara** [comment]
Yes, few things have been changed in survminer since the first release. You can read NEWS at https://github.com/kassambara/survminer/blob/master/NEWS.md

1) Now, when using ggsurvplot(), you should specify the data that have been used to fit the survival curves. Example:

```{r}
library(survival)
library(survminer)

data(lung)
fit <- survfit(Surv(time, status) ~ sex, data = lung)

ggsurvplot(fit, data = lung)

```


2) Since the version 0.4.1, the function ggcoxadjustedcurves() has been replaced by ggadjustedcurves()

3) The option to plot individual curves is not available in the new ggadjustedcurves(). I think that, after [this](https://github.com/kassambara/survminer/issues/229) discussion,  @pbiecek decided to drop this option, because taking the average of individual curves is not an optimal way to adjust survival curves. Please, make sure you have read the description of the `ggadjustedcurves()` function for the different possibilities to adjust survival curves.

**Comment by DanaiChas** [comment]
Great! Thank you for the help.

**Comment by raffaelebrustia** [comment]
Dear All, 
I'm in trouble with survminer, I got this message:
Error in ggsurvplot(fit, data = my_data) : could not find function "ggsurvplot"

I correctly installed
library(ggplot2)
library(survminer)  version 0.4.4
library(survival)

and then 

fit<- survfit(Surv(time, death) ~ 1, data = my_data)
ggsurvplot(fit, data = my_data)
Error in ggsurvplot(fit, data = my_data) : could not find function "ggsurvplot"

Just to check, I tried the basic version: 
plot(survfit(Surv(time, status)~1))
and I got a (not so nice) plot

Where I am wrong? 
Thank you for any suggestion. 

Raffaele

**Comment by kassambara** [comment]
This error is not reproducible. Please reinstall survminer

**Comment by raffaelebrustia** [comment]
Dear Kassambara, thank you for your quick answer. 
I did re-installed survminer

install.packages('survminer')
library(survminer)

and  I had this message

Error: package or namespace load failed for ‘survminer’ in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
 aucun package nommé ‘cmprsk’ n'est trouvé

So I tried with 
> install.packages("cmprsk")
Warning in install.packages :
  package ‘cmprsk’ is not available (for R version 3.5.3)

And when I try 
> ggsurvplot((survfit(Surv(time, Deces)~1)), data = my_data)
Error in ggsurvplot((survfit(Surv(time, Deces) ~ 1)),  : 
  could not find function "ggsurvplot"

Does the missing package cmprsk is the responsible?

Thank you, 

Raffaele

**Comment by raffaelebrustia** [comment]
Found the error: I hat to upgrade to version 3.6.0, and everything is going fine. 
Than you!

**Comment by Keatars** [comment]
Hi, I'm new to the survival and survmimer package and I'm trying to plot my survival curve with the ggsurvplot function. I tried this with my own data and with the example ovarian data below by running the exact steps I found online. I am using R version 3.5.1 and installed the survmimer version 0.4.4. However, running the code:

```
fit1 <- survfit(surv_object ~ rx, data = ovarian)
ggsurvplot(fit1, data = ovarian, pval = TRUE)
```

gives me the following error message:

> could not find function "ggsurvplot"

Also 

`help(ggsurvplot)`
gives me:

> No documentation for ‘ggsurvplot’ in specified packages and libraries:
> you could try ‘??ggsurvplot’

Is this function no longer supported?

Thanks for your help!

**Comment by raffaelebrustia** [comment]
Hi, 
My experience is very limited, but looks like my previous post. 
Try to reinstall an updated version of R (we are now 3.6.1), and it should work…

Good luck

**Comment by pbiecek** [comment]
@Keatars make sure that you have loaded survminer. Following lines work on a fresh  R 3.6.1

```
library("survminer")
library("survival")
fit1 <- survfit(Surv(futime , fustat) ~ rx, data = ovarian)
ggsurvplot(fit1, data = ovarian, pval = TRUE)
```

**Comment by Keatars** [comment]
Thanks a lot for your reply raffaelebrustia and pbiecek. 

I have to contact my administrator to update R for me, but it's good to know that this will most likely solve my problem!

**Comment by kassambara** [comment]
Many users have experienced an issue when trying to install the latest survminer version (v 0.4.4) from CRAN.

This is due to the recent update of the package `cmprsk`, which suddenly requires the current R version >= 3.6.0, forcing survminer users to update their R version.

As `cmprsk`  is only used in documentations, we moved it from Imports section to suggests section (in the package DESCRIPTION), so that the installation of survminer no longer needs `cmprsk`  (see https://github.com/kassambara/survminer/issues/394)


So, kindly try to install the latest dev version of survminer:

```r
devtools::install_github("kassambara/survminer", build_vignettes = FALSE)
```

**Comment by Keatars** [comment]
Thanks! Sorry for my late reply, I was unable to update it on my office computer. Works fine on my laptop though, problem solved!


Op 30 jul. 2019 om 19:34 heeft Alboukadel KASSAMBARA <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


Many users have experienced an issue when trying to install the latest survminer version (v 0.4.4) from CRAN.

This is due to the recent update of the package cmprsk, which suddenly requires the current R version >= 3.6.0, forcing survminer users to update their R version.

As cmprsk is only used in documentations, we moved it from Imports section to suggests section (in the package DESCRIPTION), so that the installation of survminer no longer needs cmprsk (see #394<https://github.com/kassambara/survminer/issues/394>)

So, kindly try to install the latest dev version of survminer:

devtools::install_github("kassambara/survminer", build_vignettes = FALSE)

**Comment by mattiafonderico** [comment]
Hi everybody.
I followed this conversation cause I found a similar  issue.  
R version: 4.0.5; I installed "survminer" a couple of weeks ago.
I performed a multivariate cox regression model:
```
library(survival)
library(survminer)
cox.mod1 <- coxph(Surv(Timeto7Yr,EDSS7)~ Sex + Age + Diseaseduration + `Visit density` + FEDSS + Prerelapses + Superimposed * DMT, data = Database_solo_covariate_1_139)```

Now, I faced two problems with the survminer package:
1) I can not find anywhere the function "ggcoxadjustedcurves"
2) I founnd the function "ggsurvplot" but when I run it, R gives me the following error:

```
> ggsurvplot(cox.mod1, data)
Error in ggsurvplot(cox.mod1) : oggetto "ggsurv" non trovato
```

Can anybody help me?
